File: android-core/src/androidTest/java/com/mparticle/internal/database/tables/UploadTableTest.java
package com.mparticle.internal.database.tables;

import android.database.sqlite.SQLiteDatabase;

import com.mparticle.internal.database.services.SQLiteOpenHelperWrapper;

import org.junit.Test;

public class UploadTableTest extends BaseTableTest{
    @Test
    public void createTableTest() throws InterruptedException {
        runTest(new SQLiteOpenHelperWrapper() {
            @Override
            public void onCreate(SQLiteDatabase database) {
                database.execSQL(UploadTable.CREATE_UPLOADS_DDL);
            }

            @Override
            public void onUpgrade(SQLiteDatabase database, int oldVersion, int newVersion) {
                //do nothing
            }

            @Override
            public void onDowngrade(SQLiteDatabase database, int oldVersion, int newVersion) {

            }
        });
    }
}

File: android-core/src/androidTest/java/com/mparticle/internal/database/tables/UploadTableTest.kt
package com.mparticle.internal.database.tables

import android.database.sqlite.SQLiteDatabase
import com.mparticle.internal.database.services.SQLiteOpenHelperWrapper
import org.junit.Test

class UploadTableTest : BaseTableTest() {
    @Test
    @Throws(InterruptedException::class)
    fun createTableTest() {
        runTest(object : SQLiteOpenHelperWrapper {
            override fun onCreate(database: SQLiteDatabase) {
                database.execSQL(UploadTable.CREATE_UPLOADS_DDL)
            }

            override fun onUpgrade(database: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
                // do nothing
            }

            override fun onDowngrade(database: SQLiteDatabase, oldVersion: Int, newVersion: Int) {}
        })
    }
}
